v2/yarpcfx/yarpcfx.go

```diff
@@ -191,8 +193,31 @@ func NewRouter(p RouterParams) (RouterResult, error) {
 	for _, pl := range p.ProcedureLists {
 		procedures = append(procedures, pl...)
 	}
-	router := yarpcrouter.NewMapRouter("foo" /* Derive from servicefx. */, procedures)
+
+	proceduresWithMW := proceduresWithMiddleware(procedures, p.UnaryTransportMiddleware)
+	router := yarpcrouter.NewMapRouter("foo" /* Derive from servicefx. */, proceduresWithMW)
+
 	return RouterResult{
 		Router: yarpc.ApplyRouter(router, p.RouterMiddleware),
 	}, nil
 }
+
+// proceduresWithMiddleware applies middleware to the procedures
+func proceduresWithMiddleware(
```

Comment:
optional: `applyMiddlewareToProcedures` might be more consistent with the naming above.